src/nvim/ops.c

```diff
@@ -915,24 +915,43 @@ int do_record(int c)
       apply_autocmds(EVENT_RECORDINGENTER, NULL, NULL, false, curbuf);
     }
   } else {  // stop recording
+    save_v_event_T save_v_event;
+    // Set the v:event dictionary with information about the recording.
+    dict_T *dict = get_v_event(&save_v_event);
+
+    // The recorded text contents.
+    list_T *const list = tv_list_alloc(1);
+    p = get_recorded();
+    if (p != NULL) {
+      // Remove escaping for CSI and K_SPECIAL in multi-byte chars.
+      vim_unescape_csi(p);
+
+      tv_list_append_string(list, (const char *)p, -1);
```

Comment:
Why use a list for `regcontents`?

Reply:
Maybe unnecessary indeed but I did it like this to be consistent with eg `TextYankPost`.

Reply:
Hmm, `TextYankPost` does it that way because you can yank multiple lines and it puts them in one line per list entry. I am not sure that makes sense for a macro recording however. But still, the consistency might be good.

I would lean toward making it just a string, but curious what @bfredl thinks.

Reply:
well a register needs the linelist as it could contain NUL. I don't think a recording ever will contain NUL so it is probably fine.

Reply:
I don't have a very strong opinion here, do I understand you both that you'd be fine with going with a plain string?

Reply:
Yes I think so.

Reply:
Changed to string in 28d2ce10aebd23a8cc94a38a1835edeeeb50c7fc